test/functional/eval/has_spec.lua

```diff
@@ -0,0 +1,51 @@
+local helpers = require('test.functional.helpers')(after_each)
+local eq = helpers.eq
+local clear = helpers.clear
+local funcs = helpers.funcs
+
+describe('has()', function()
+  before_each(clear)
+
+  it('"nvim-x.y.z"', function()
+    eq(0, funcs.has("nvim-"))
+    eq(0, funcs.has("nvim-  "))
+    eq(0, funcs.has("nvim- \t "))
+    eq(0, funcs.has("nvim-0. 1. 1"))
+    eq(0, funcs.has("nvim-0. 1.1"))
+    eq(0, funcs.has("nvim-0.1. 1"))
+    eq(0, funcs.has("nvim-a"))
+    eq(0, funcs.has("nvim-a.b.c"))
+    eq(0, funcs.has("nvim-0.b.c"))
+    eq(0, funcs.has("nvim-0.0.c"))
+    eq(0, funcs.has("nvim-0.b.0"))
+    eq(0, funcs.has("nvim-a.b.0"))
+    eq(0, funcs.has("nvim-0.1"))
```

Comment:
That's because of the `strlen(name) >= 10` check in `f_has()`. I originally did it that way because that's how `has('patch-x.y.zzzz')` works, but given that `has_nvim_version()` searches for dots, we don't really need that restrction. I'll change it to be more permissive.